### PR TITLE
Add missing `region` and `showNonLocalizedStrings` to `RunActionOptions` related types

### PR DESCRIPTION
**Pull request by ladislas**
Resolves https://github.com/tuist/tuist/discussions/5781

### Short description 📝

- fix missing `region`
- add `showNonLocalizedStrings` to RunActionOptions

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by ladislas** [comment]
@pepicrft following our discussion here: https://github.com/tuist/tuist/discussions/5781

It seems like I also need to do a PR on xcodeproj to update XCScheme, I'll work on that now

**Comment by danyf90** [comment]
Please follow PR template

**Comment by ladislas** [comment]
> I have updated the PR title, next time please follow PR template

Yes sorry @danyf90, you were too quick ;) I'm using `gh` to create my PRs and it skipped the template. It also uses the branch name for the title if there is only one commit.

I've updated the description.

**Comment by ladislas** [comment]
@pepicrft @danyf90 XcodeProj PR here: https://github.com/tuist/XcodeProj/pull/806

File: Sources/ProjectDescription/RunActionOptions.swift [hunk 1]
```diff
@@ -8,6 +8,9 @@ public struct RunActionOptions: Equatable, Codable {
     /// Region to use when running the app.
     public var region: String?
 
+    /// Show non localized string as UPPERCASE when running the app.
+    public var showNonLocalizedStrings: Bool
```
Comment (danyf90) [inline Sources/ProjectDescription/RunActionOptions.swift:12]:
```suggestion
    /// Show non localized string as UPPERCASE when running the app.
    public var showNonLocalizedStringsAsUppercase: Bool
```

File: Sources/TuistGraph/Models/RunActionOptions.swift [hunk 2]
```diff
@@ -9,6 +9,9 @@ public struct RunActionOptions: Equatable, Codable {
     /// App Region.
     public let region: String?
 
+    /// Show non localized string as UPPERCASE when running the app.
+    public var showNonLocalizedStrings: Bool
```
Comment (danyf90) [inline Sources/TuistGraph/Models/RunActionOptions.swift:13]:
```suggestion
    public var showNonLocalizedStringsAsUppercase: Bool
```

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift [hunk 3]
```diff
@@ -13,11 +13,15 @@ extension TuistGraph.RunActionOptions {
         generatorPaths: GeneratorPaths
```
Comment (pepicrft) [inline Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift:13]: Do you mind creating a test file similar [to this one](https://github.com/tuist/tuist/blob/main/Tests/TuistLoaderTests/Models%2BManifestMappers/CodeCoverageMode%2BManifestMapperTests.swift) where we test that the properties are properly mapped?

File: Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift [hunk 4]
```diff
@@ -13,11 +13,15 @@ extension TuistGraph.RunActionOptions {
...
         var language: String?
+        var region: String?
+        var showNonLocalizedStringsAsUppercase: Bool
         var storeKitConfigurationPath: AbsolutePath?
         var simulatedLocation: SimulatedLocation?
         var enableGPUFrameCaptureMode: GPUFrameCaptureMode
 
         language = manifest.language?.identifier
+        region = manifest.region
+        showNonLocalizedStringsAsUppercase = manifest.showNonLocalizedStringsAsUppercase
```
Comment (pepicrft) [inline Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift:24]: Note that this is not sufficient, you still need to adjust [this logic](https://github.com/tuist/tuist/blob/78367e26028506407c393c657580ba30386fa0d4/Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift#L396) to set the right value in the generated scheme.

Comment (ladislas) [reply Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift:24]: you mean when XcodeProj has been updated? If so, yes I know. If not, then I'm not sure I understand :)

**Comment by ladislas** [comment]
@pepicrft I've added the tests, but I must confess that your test architecture is a bit cryptic and thus I'm not sure if I'm testing correctly or even testing anything at all.

Feel free to correct me and/or guide me in the right direction if needed :)

File: Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift (hunk 4 above)
Comment (pepicrft) [reply Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift:24]: There's a bit of indirection that might explain the confusion. The following mappings happen in the process of generating a project.
- `RunActionOptions (ProjectDescription)` to `RunActionOptions (TuistGraph)`: This maps the public models from ProjectDescriptiont to the ones that Tuist uses internally, which are slightly different from the public ones. For example, the internal ones have the glob patterns resolved.
- `RunActionOptions (TuistGraph)` to `XCScheme.LaunchAction`: This maps the internal model to the actual representation in the generated Xcode project.

The **later is missing**. You'll have to update XcodeProj to the latest version, which I have just released, and then update [the logic here](https://github.com/tuist/tuist/blob/78367e26028506407c393c657580ba30386fa0d4/Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift#L603) to include the new attribute. Let me know if it's clearer now.

File: Sources/ProjectDescription/RunActionOptions.swift [hunk 5]
```diff
@@ -8,6 +8,9 @@ public struct RunActionOptions: Equatable, Codable {
     /// Region to use when running the app.
     public var region: String?
 
+    /// Show non localized string as UPPERCASE when running the app.
+    public var showNonLocalizedStringsAsUppercase: Bool
```
Comment (pepicrft) [inline Sources/ProjectDescription/RunActionOptions.swift:12]: What is the `AsUppercase` for? I see [XCScheme.LaunchAction](https://github.com/tuist/XcodeProj/commit/2b9c2acfbabc87649d65486b8171c83cbccc18b7#diff-e034a7d7b7a01c4e1b3d83eb419ecf1e9c6cfb76e7fe9ed2e812222f4149c572R68) calls it `showNonLocalizedStrings`, so I'd stick to that naming.

Comment (ladislas) [reply Sources/ProjectDescription/RunActionOptions.swift:12]: this was a suggestion by @danyf90 
I'll revert the change

File: Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift (hunk 4 above)
Comment (ladislas) [reply Sources/TuistLoader/Models+ManifestMappers/RunActionOptions+ManifestMapper.swift:24]: working on it now :)

File: Sources/ProjectDescription/RunActionOptions.swift (hunk 5 above)
Comment (pepicrft) [reply Sources/ProjectDescription/RunActionOptions.swift:12]: What's the effect of `showNonLocalizedStrings` in the project? I was in favor of consistency, but if the name is not representative enough, the `AsUppercase` bit might make sense.

Comment (ladislas) [reply Sources/ProjectDescription/RunActionOptions.swift:12]: on the project nothing, but when running the app, strings that are not localized are UPPERCASE, to be easier to spot.

Comment (ladislas) [reply Sources/ProjectDescription/RunActionOptions.swift:12]: IMHO the fact that it's uppercase is an implementation details that only apple can control. if they decide in their next xcode update to make the text pink and blinking, we should not have to change the name of the option

Comment (pepicrft) [reply Sources/ProjectDescription/RunActionOptions.swift:12]: Alright, in that case remove the uppercase from the name.

**Comment by mihaicris-adoreme** [comment]
Missing `region` has been added in #5954

**Comment by philipbel** [comment]
Any reason the `showNonLocalizedStrings` flag was not merged?

**Pull request closed without merging**